### PR TITLE
chore: throw amplify error for headless pull and headless import auth

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
-  throw amplify error for headless pull and headless import auth when missing parameters passed from script or env variables
- Added same for amplify import storage headless

#### Description of how you validated changes
- Added unit tests 
- Manual testing in headless pull to check AmplifyError is thrown.
#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/index.ts [hunk 1]
```diff
@@ -155,7 +155,10 @@ export const updateConfigOnEnvInit = async (context: $TSContext, category: any,
       });
 
       if (missingParams.length) {
-        throw new Error(`auth headless is missing the following inputParams ${missingParams.join(', ')}`);
+        throw new AmplifyError( 'InputValidationError', {
```
Comment (akshbhu) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/index.ts:158]: only change

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if (!headlessParams.nativeClientId) {
+      missingParams.push('nativeClientId');
+    }
+  
+    if (resourceParameters.authSelections === 'identityPoolAndUserPool' && !headlessParams.identityPoolId) {
+      missingParams.push('identityPoolId');
+    }
+  
+    if (missingParams.length > 0) {
+      throw new AmplifyError( 'InputValidationError', {
```
Comment (akshbhu) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts:31]: same here

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        message: `auth headless is missing the following inputParameters ${missingParams.join(', ')}`,
+        link: 'https://docs.amplify.aws/cli/usage/headless/#--categories'
+      }); 
+    }
+  
+    const envSpecificParameters: EnvSpecificResourceParameters = {
+      userPoolId: headlessParams.userPoolId,
+      userPoolName: '', // Will be filled out later
```
Comment (sobolk) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts:39]: is it mandatory parameter ?

**Review by sobolk — COMMENTED** [review]
Please address codeql warnings.

**Review by lazpavel — APPROVED** [review]
LGTM, there is new warning introduced

**Review by sobolk — APPROVED** [review]

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts (hunk 3 above)
Comment (akshbhu) [reply packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts:39]: It is not , that why I guess will be filled out later :)
May be used here  to share the Type def

Comment (sobolk) [reply packages/amplify-category-auth/src/provider-utils/awscloudformation/import/ensure-headless-parameters.ts:39]: should be typed as `userPoolName?: string;` but that's out of scope of this PR, unless you want a challenge ;-)

**Review by jhockett — APPROVED** [review]

**Review by lazpavel — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

**Pull request merged**